app/controllers/work_packages/moves_controller.rb

```diff
@@ -114,8 +115,9 @@ def prepare_for_work_package_move
     @allowed_projects = WorkPackage.allowed_target_projects_on_move(current_user)
     @target_project = @allowed_projects.detect { |p| p.id.to_s == params[:new_project_id].to_s } if params[:new_project_id]
     @target_project ||= @project
-    @types = @target_project.types
+    @types = @target_project.types.order(:position)
     @target_type = @types.find { |t| t.id.to_s == params[:new_type_id].to_s }
+    @unavailable_type_in_target_project = @types.exclude?(@target_type)
```

Comment:
If the `@target_project == @project` then the `@unavailable_type_in_target_project` should be `false`.

Furthermore, when I move a WP with type task, that is enabled in the target project too while leaving the `Type` on `(No change)`, the `params[:new_type_id]` will not be set, which in turn makes the warning to be shown. That is incorrect.

Reply:
Nice catch. I understand now what you mean

Addressed in latest commit!